pkg/services/sqlstore/team.go

```diff
@@ -91,6 +107,16 @@ func (ss *SQLStore) CreateTeam(name, email string, orgID int64) (models.Team, er
 		_, err := sess.Insert(&team)
 		return err
 	})
+	// In PostgreSQL the sequence is not updated on explicit INSERTs with a provided value for an autoincremented column
+	// However, after the failure, the sequence is set to the current value; therefore, a subsequent INSERT succeeds.
+	// https://gist.github.com/zserge/694b0a1dfb76a7366d5000cda93ee2e5
```

Comment:
<details>
<summary>Example of PostgreSQL sequence behaviour</summary>

```
~/go/src/github.com/grafana/grafana/devenv: docker compose exec postgrestest psql -U grafanatest
psql (9.3.25)
Type "help" for help.

grafanatest=# SELECT last_value FROM team_id_seq;
 last_value 
------------
          2
(1 row)

grafanatest=# INSERT INTO team(id, name, org_id, created, updated, email) VALUES (3, 'just-testing', 1, NOW(), NOW(), '');
INSERT 0 1
grafanatest=# SELECT last_value FROM team_id_seq;
 last_value 
------------
          2
(1 row)

grafanatest=# INSERT INTO team(name, org_id, created, updated, email) VALUES ('just-testing-2', 1, NOW(), NOW(), '');
ERROR:  duplicate key value violates unique constraint "team_pkey"
DETAIL:  Key (id)=(3) already exists.
grafanatest=# SELECT last_value FROM team_id_seq;
 last_value 
------------
          3
(1 row)

grafanatest=# INSERT INTO team(name, org_id, created, updated, email) VALUES ('just-testing-2', 1, NOW(), NOW(), '');
INSERT 0 1
grafanatest=# SELECT last_value FROM team_id_seq;
 last_value 
------------
          4
(1 row)

grafanatest=# 
```
</details>

Reply:
Also notice [this check](https://github.com/grafana/grafana/blob/65d7d466d781e2183041a46b7881e1e574ab50e0/pkg/services/sqlstore/team.go#L85) that seems redundant given (there is [a unique index](https://github.com/grafana/grafana/blob/65d7d466d781e2183041a46b7881e1e574ab50e0/pkg/services/sqlstore/migrations/team_mig.go#L17) on team org and name)